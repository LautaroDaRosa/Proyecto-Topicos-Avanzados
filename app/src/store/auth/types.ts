import { API_STATUS } from 'utils/constants';

export interface LoginDataType {
  userId: number;
  username: string;
  phone: string;
  email: string;
  info: string;
  logo: string;
  role: string;
  token: string;
}

export interface RegisterDataType {
  username: string;
  phone: string;
  email: string;
  info: string;
  logo: string;
  rut: string;
  address: string;
  contact: string;
  businessName: string;
}

export interface AuthState {
  login: {
    error: string | number | undefined;
    status: API_STATUS | undefined;
  };
}
