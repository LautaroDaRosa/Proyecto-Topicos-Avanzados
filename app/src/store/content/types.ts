import { API_STATUS } from 'utils/constants';

export interface PostsState {
  postsApiResult: {
    posts: Post[] | undefined;
    status: API_STATUS | undefined;
  };
}

export interface PostsType {
  posts: PostType[];
}

export interface PostType {
  posts: number;
  email: string;
  fullName: string;
  firstName: string;
  lastName: string;
}

export interface Post {
  id: number;
  user: {
    id: number;
    fullname: string;
    photo: string;
  };
  content: string;
  photo: string;
  createdAt: string;
}
