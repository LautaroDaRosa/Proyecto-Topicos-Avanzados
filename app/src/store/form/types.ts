export interface QuestionAnswer {
  question: number;
  response: number;
}

export interface ListQuestionId {
  listQuestionId: number[];
}
