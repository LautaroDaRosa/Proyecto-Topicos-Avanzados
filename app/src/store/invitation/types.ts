export interface Invitation {
  inviteSender: {
    userId: number;
    username: string;
    phone: string;
    email: string;
    info: string;
    logo: string;
    role: string;
  };
  invteReceiver: string;
}
