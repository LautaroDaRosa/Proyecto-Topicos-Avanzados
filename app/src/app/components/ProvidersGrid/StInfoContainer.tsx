import styled from 'styled-components';

const StInfoContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  button {
    margin-top: 12px;
  }
`;

export default StInfoContainer;
