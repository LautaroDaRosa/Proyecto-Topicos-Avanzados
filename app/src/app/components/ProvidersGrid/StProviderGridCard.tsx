import styled from 'styled-components';

const StProviderGridCard = styled.div`
  display: flex;
  background: white;
  border-radius: 8px;
  padding: 24px;
  box-shadow: 0px 0px 20px 0px lightgrey;
  flex: 1;
  align-items: center;
  flex-direction: column;
  justify-content: space-between;

  img {
    max-width: 180px;
    max-height: 140px;
    margin: 16px;
  }
`;

export default StProviderGridCard;
