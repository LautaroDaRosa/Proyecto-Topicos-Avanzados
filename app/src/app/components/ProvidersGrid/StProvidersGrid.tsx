import styled from 'styled-components';
import { media } from 'styles/media';

const StProvidersGrid = styled.div`
  display: grid;
  width: 100%;
  gap: 16px;

  grid-template-columns: 1fr;

  ${media.medium} {
    grid-template-columns: 1fr 1fr 1fr;'
  }`;

export default StProvidersGrid;
