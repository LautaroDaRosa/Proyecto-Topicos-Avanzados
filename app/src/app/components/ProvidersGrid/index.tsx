import { Provider } from 'store/providers/types';
import StProviderGridCard from './StProviderGridCard';
import StProvidersGrid from './StProvidersGrid';
import Button from '../Button';
import StInfoContainer from './StInfoContainer';

interface Providers {
  providers: Provider[];
}

const ProvidersGrid = ({ providers }: Providers) => {
  return (
    <StProvidersGrid>
      {providers.map(provider => (
        <StProviderGridCard key={provider.userId}>
          <h3>{provider.name}</h3>
          <img src={provider.logo} alt={`${provider.name} Logo`} />
          <StInfoContainer>
            <span>Score: {provider.averageScore}</span>
            <span>Razon Social: {provider.businessName}</span>
            <span>RUT: {provider.rut}</span>
            <Button
              disabled={true}
              text="Ver perfil de proveedor"
              action="primary"
            />
          </StInfoContainer>
        </StProviderGridCard>
      ))}
    </StProvidersGrid>
  );
};

export default ProvidersGrid;
