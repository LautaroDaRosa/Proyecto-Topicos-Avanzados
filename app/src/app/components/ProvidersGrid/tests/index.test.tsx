import * as React from 'react';
import { render } from '@testing-library/react';

import ProvidersGrid from '..';
import { Provider } from 'store/providers/types';

const mockedProviders: Provider[] = [
  {
    name: 'Canal 10',
    businessName: 'CANAL 10 S.A.',
    rut: '734879753738964',
    logo: 'https://upload.wikimedia.org/wikipedia/commons/thumb/c/c8/Canal_10_Uruguay_Logo_1960.webp/640px-Canal_10_Uruguay_Logo_1960.webp.png',
    userId: 1,
    averageScore: 9,
  },
  {
    name: 'Canal 4',
    businessName: 'CANAL 4 S.A.',
    rut: '633334234363643',
    logo: 'https://upload.wikimedia.org/wikipedia/commons/3/33/Canal4_uy.png',
    userId: 2,
    averageScore: 6,
  },
];

describe('Filter Bar', () => {
  it('should match snapshot', () => {
    const providersGrid = render(<ProvidersGrid providers={mockedProviders} />);
    expect(providersGrid).toMatchSnapshot();
  });
  it('should show all the providers information', () => {
    const providersGrid = render(<ProvidersGrid providers={mockedProviders} />);
    providersGrid.getByText('Canal 10');
    providersGrid.getByText('Razon Social: CANAL 10 S.A.');
    providersGrid.getByText('RUT: 734879753738964');
    providersGrid.getByText('Score: 9');

    providersGrid.getByText('Canal 4');
    providersGrid.getByText('Razon Social: CANAL 4 S.A.');
    providersGrid.getByText('RUT: 633334234363643');
    providersGrid.getByText('Score: 6');
  });
});
