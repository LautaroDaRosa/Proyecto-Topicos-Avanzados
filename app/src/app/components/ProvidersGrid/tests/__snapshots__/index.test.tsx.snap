// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Bar should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: white;
  border-radius: 8px;
  padding: 24px;
  box-shadow: 0px 0px 20px 0px lightgrey;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 img {
  max-width: 180px;
  max-height: 140px;
  margin: 16px;
}

.c0 {
  display: grid;
  width: 100%;
  gap: 16px;
  grid-template-columns: 1fr;
}

.c3 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  border: 1.5px solid #fca408;
  background-color: white;
  color: #fca408;
}

.c3:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c3:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #fca408;
}

.c3:active:enabled {
  color: white;
  background-color: #b27508;
  border-color: #b27508;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 button {
  margin-top: 12px;
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <h3>
            Canal 10
          </h3>
          <img
            alt="Canal 10 Logo"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c8/Canal_10_Uruguay_Logo_1960.webp/640px-Canal_10_Uruguay_Logo_1960.webp.png"
          />
          <div
            class="c2"
          >
            <span>
              Score: 
              9
            </span>
            <span>
              Razon Social: 
              CANAL 10 S.A.
            </span>
            <span>
              RUT: 
              734879753738964
            </span>
            <button
              action="primary"
              class="c3"
              disabled=""
            >
              Ver perfil de proveedor
            </button>
          </div>
        </div>
        <div
          class="c1"
        >
          <h3>
            Canal 4
          </h3>
          <img
            alt="Canal 4 Logo"
            src="https://upload.wikimedia.org/wikipedia/commons/3/33/Canal4_uy.png"
          />
          <div
            class="c2"
          >
            <span>
              Score: 
              6
            </span>
            <span>
              Razon Social: 
              CANAL 4 S.A.
            </span>
            <span>
              RUT: 
              633334234363643
            </span>
            <button
              action="primary"
              class="c3"
              disabled=""
            >
              Ver perfil de proveedor
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: white;
  border-radius: 8px;
  padding: 24px;
  box-shadow: 0px 0px 20px 0px lightgrey;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 img {
  max-width: 180px;
  max-height: 140px;
  margin: 16px;
}

.c0 {
  display: grid;
  width: 100%;
  gap: 16px;
  grid-template-columns: 1fr;
}

.c3 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  border: 1.5px solid #fca408;
  background-color: white;
  color: #fca408;
}

.c3:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c3:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #fca408;
}

.c3:active:enabled {
  color: white;
  background-color: #b27508;
  border-color: #b27508;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 button {
  margin-top: 12px;
}

@media (min-width:768px) {
  .c0 {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <h3>
          Canal 10
        </h3>
        <img
          alt="Canal 10 Logo"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c8/Canal_10_Uruguay_Logo_1960.webp/640px-Canal_10_Uruguay_Logo_1960.webp.png"
        />
        <div
          class="c2"
        >
          <span>
            Score: 
            9
          </span>
          <span>
            Razon Social: 
            CANAL 10 S.A.
          </span>
          <span>
            RUT: 
            734879753738964
          </span>
          <button
            action="primary"
            class="c3"
            disabled=""
          >
            Ver perfil de proveedor
          </button>
        </div>
      </div>
      <div
        class="c1"
      >
        <h3>
          Canal 4
        </h3>
        <img
          alt="Canal 4 Logo"
          src="https://upload.wikimedia.org/wikipedia/commons/3/33/Canal4_uy.png"
        />
        <div
          class="c2"
        >
          <span>
            Score: 
            6
          </span>
          <span>
            Razon Social: 
            CANAL 4 S.A.
          </span>
          <span>
            RUT: 
            633334234363643
          </span>
          <button
            action="primary"
            class="c3"
            disabled=""
          >
            Ver perfil de proveedor
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
